tests/test_transpiler_pipeline.py

```diff
@@ -127,40 +146,51 @@ def test_assert_circuit_equivalence_wrong_nqubits():
 
 def test_error_connectivity():
     with pytest.raises(TranspilerPipelineError):
-        default_transpiler = Passes()
+        default_transpiler = Passes(passes=None, connectivity=None)
 
 
 def test_is_satisfied():
-    default_transpiler = Passes(connectivity=star_connectivity())
+    default_transpiler = Passes(passes=None, connectivity=star_connectivity())
     circuit = Circuit(5)
     circuit.add(gates.CZ(0, 2))
     circuit.add(gates.Z(0))
     assert default_transpiler.is_satisfied(circuit)
 
 
 def test_is_satisfied_false_decomposition():
-    default_transpiler = Passes(connectivity=star_connectivity())
+    default_transpiler = Passes(passes=None, connectivity=star_connectivity())
     circuit = Circuit(5)
     circuit.add(gates.CZ(0, 2))
     circuit.add(gates.X(0))
     assert not default_transpiler.is_satisfied(circuit)
 
 
 def test_is_satisfied_false_connectivity():
-    default_transpiler = Passes(connectivity=star_connectivity())
+    default_transpiler = Passes(passes=None, connectivity=star_connectivity())
     circuit = Circuit(5)
     circuit.add(gates.CZ(0, 1))
     circuit.add(gates.Z(0))
     assert not default_transpiler.is_satisfied(circuit)
 
 
+@pytest.mark.parametrize("reps", [range(3)])
+@pytest.mark.parametrize("placer", [Random, Trivial, ReverseTraversal])
+@pytest.mark.parametrize("routing", [ShortestPaths, Sabre])
 @pytest.mark.parametrize(
     "circ", [generate_random_circuit(nqubits=5, ngates=20), small_circuit()]
 )
-def test_custom_passes(circ):
+def test_custom_passes(circ, placer, routing, reps):
     custom_passes = []
     custom_passes.append(Preprocessing(connectivity=star_connectivity()))
-    custom_passes.append(Random(connectivity=star_connectivity()))
+    if placer == ReverseTraversal:
+        custom_passes.append(
+            placer(
+                connectivity=star_connectivity(),
+                routing_algorithm=routing(connectivity=star_connectivity()),
+            )
+        )
+    else:
+        custom_passes.append(placer(connectivity=star_connectivity()))
     custom_passes.append(ShortestPaths(connectivity=star_connectivity()))
```

Comment:
Should this be testing both sabre and shortestpaths?
```suggestion
    custom_passes.append(routing(connectivity=star_connectivity()))
```